src/Catalyst.Core.Lib.Tests/UnitTests/P2P/IO/Observers/GetNeighbourResponseObserverTests.cs

```diff
@@ -0,0 +1,103 @@
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using Catalyst.Abstractions.P2P.IO.Messaging.Dto;
+using Catalyst.Core.Lib.Extensions;
+using Catalyst.Core.Lib.IO.Messaging.Correlation;
+using Catalyst.Core.Lib.P2P.IO.Observers;
+using Catalyst.Protocol.IPPN;
+using Catalyst.Protocol.Peer;
+using Catalyst.TestUtils;
+using DotNetty.Transport.Channels;
+using Google.Protobuf;
+using Microsoft.Reactive.Testing;
+using NSubstitute;
+using Serilog;
+using Xunit;
+
+namespace Catalyst.Core.Lib.Tests.UnitTests.P2P.IO.Observers
+{
+    public sealed class GetNeighbourResponseObserverTests : IDisposable
+    {
+        private readonly IChannelHandlerContext _fakeContext;
+        private readonly GetNeighbourResponseObserver _observer;
+
+        public GetNeighbourResponseObserverTests()
+        {
+            _fakeContext = Substitute.For<IChannelHandlerContext>();
+            _observer = new GetNeighbourResponseObserver(Substitute.For<ILogger>());
+        }
+
+        [Fact]
+        public void Observer_Can_Process_GetNeighbourResponse_Correctly()
+        {
+            var testScheduler = new TestScheduler();
+
+            var peers = new[]
+            {
+                PeerIdHelper.GetPeerId(),
+                PeerIdHelper.GetPeerId(),
+                PeerIdHelper.GetPeerId()
+            };
+
+            var peerNeighborsResponse = new PeerNeighborsResponse
+            {
+                Peers =
+                {
+                    peers
+                }
+            };
+            var protocolMessage =
+                peerNeighborsResponse.ToProtocolMessage(PeerIdHelper.GetPeerId("sender"),
+                    CorrelationId.GenerateCorrelationId());
+
+            var peerNeighborsResponseObserver = Substitute.For<IObserver<IPeerClientMessageDto>>();
+
+            var messageStream = MessageStreamHelper.CreateStreamWithMessage(_fakeContext, testScheduler,
+                protocolMessage);
+
+            _observer.StartObserving(messageStream);
+
+            using (_observer.MessageStream.Subscribe(peerNeighborsResponseObserver.OnNext))
+            {
+                testScheduler.Start();
+
+                peerNeighborsResponseObserver.Received(1)
+                   .OnNext(Arg.Is<IPeerClientMessageDto>(p => test(p.Message, peers[0])));
+                peerNeighborsResponseObserver.Received(1)
+                   .OnNext(Arg.Is<IPeerClientMessageDto>(p => test(p.Message, peers[1])));
+                peerNeighborsResponseObserver.Received(1)
+                   .OnNext(Arg.Is<IPeerClientMessageDto>(p => test(p.Message, peers[2])));
+            }
+        }
+
+        private bool test(IMessage msg, PeerId peerId)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Rename method 'test' to match pascal case naming rules, consider using 'Test'.](https://app.codacy.com/app/catalyst-network/Catalyst.Node/pullRequest?prid=4318201)